src/java/picard/illumina/CollectIlluminaLaneMetrics.java

```diff
@@ -95,15 +101,10 @@ public static void main(final String[] args) {
             } catch (final FileNotFoundException e) {
                 throw new PicardException("Unable to open laneMetrics file.", e);
             }
-    
-            return CollectionUtil.partition(tiles,
-                    new CollectionUtil.Partitioner<Tile, Integer>() {
-                        @Override
-                        public Integer getPartition(final Tile tile) {
-                            return tile.getLaneNumber();
-                        }
-                    });
-        }
+
+            Map<Integer,? extends Collection<Tile>> tilesByLane=tiles.stream().collect(Collectors.groupingBy(Tile::getLaneNumber));
```

Comment:
final needed.  Actually, just merge these into a single line please - you don't need a local variable.